### PR TITLE
API-93: Check API overall access

### PR DESCRIPTION
**Pull request by phaseinducer**
This PR adds an access control on the `/api/` URL pattern. It ensures that users have the "overrall_api_access" ACL granted for their role.

Since it's the first time (I guess) that we need to check ACLs not linked to any Oro annotation so early in the app (we already did but not inside the security itself) some modifications were needed inside Oro ACL implementation.
See the description in the new voter to have more info.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/tests/integration/Security/AuthorizationIntegration.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace tests\integration\Pim\Bundle\ApiBundle\Security;
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Security/AuthorizationIntegration.php]: namespace Pim\Bundle\ApiBundle\tests\integration\Security

File: app/config/security.yml [hunk 2]
```diff
@@ -66,4 +66,4 @@ security:
 
     access_control:
         - { path: ^/admin/, role: ROLE_ADMIN }
-        - { path: ^/api/, role: IS_AUTHENTICATED_FULLY }
+        - { path: ^/api/, role: pim_api_overall_access }
```
Comment (skeleton) [inline app/config/security.yml]: the list of routes:
```
 oro_user_apigen                                 ANY      ANY    ANY  /user/apigen/{id}                                                          
 oro_api_get_navigationitems                     GET      ANY    ANY  /api/rest/{version}/navigationitems/{type}.{_format}                       
 oro_api_post_navigationitems                    POST     ANY    ANY  /api/rest/{version}/navigationitems/{type}.{_format}                       
 oro_api_put_navigationitems_id                  PUT      ANY    ANY  /api/rest/{version}/navigationitems/{type}/ids/{itemId}.{_format}          
 oro_api_delete_navigationitems_id               DELETE   ANY    ANY  /api/rest/{version}/navigationitems/{type}/ids/{itemId}.{_format}          
 oro_api_get_pagestates                          GET      ANY    ANY  /api/rest/{version}/pagestates.{_format}                                   
 oro_api_get_pagestate                           GET      ANY    ANY  /api/rest/{version}/pagestates/{id}.{_format}                              
 oro_api_post_pagestate                          POST     ANY    ANY  /api/rest/{version}/pagestates.{_format}                                   
 oro_api_put_pagestate                           PUT      ANY    ANY  /api/rest/{version}/pagestates/{id}.{_format}                              
 oro_api_delete_pagestate                        DELETE   ANY    ANY  /api/rest/{version}/pagestates/{id}.{_format}                              
 oro_api_get_pagestate_checkid                   GET      ANY    ANY  /api/rest/{version}/pagestate/checkid.{_format} 
``` 

it concerns the oro navigation, dunno if it's still used :(

Comment (skeleton) [inline app/config/security.yml]: I removed src/Oro/Bundle/NavigationBundle/Controller/Api/NavigationItemController & src/Oro/Bundle/NavigationBundle/Controller/Api/PagestateController.php to test the application, it seems to work but I admit I don't know how work the amazing feature "pinbar"

Comment (phaseinducer) [inline app/config/security.yml]: I changed Oro URLs, it works :+1:

Comment (a2xchip) [inline app/config/security.yml]: :-)

**Pull request merged**